src/main/java/com/puppetlabs/jenkins/plugins/puppetgatling/SimulationMetrics.java

```diff
@@ -0,0 +1,81 @@
+package com.puppetlabs.jenkins.plugins.puppetgatling;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.puppetlabs.jenkins.plugins.puppetgatling.chart.Graph;
+import com.puppetlabs.jenkins.plugins.puppetgatling.chart.RawDataGraph;
+import hudson.FilePath;
+import hudson.model.Run;
+import io.gatling.jenkins.chart.Point;
+import io.gatling.jenkins.chart.Serie;
+import io.gatling.jenkins.chart.SerieName;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class SimulationMetrics {
+
+    private static final int MAX_MEMORY_DATA_POINTS_TO_DISPLAY = 40;
+    private static final int BYTES_PER_MEGABYTE = (1024 * 1024);
+
+    private final Graph<Long> memoryUsage;
+
+    public SimulationMetrics(Run<?, ?> run, PrintStream logger, FilePath workspace,
+                             String simulationId) throws IOException, InterruptedException {
+
+        logger.println("Checking for existence of metrics.json.");
+        FilePath metricsFilePath = workspace.child("puppet-gatling").
+                child(simulationId).child("sut_archive_files").
+                child("metrics.json");
+        if (metricsFilePath.exists()) {
+            logger.println("Found metrics.json, parsing.");
+            List<Point<Integer, Long>> memoryData = new ArrayList<>();
+
+            InputStream metricsInput = metricsFilePath.read();
+            JsonFactory jsonFactory = new JsonFactory();
+            JsonParser jsonParser = jsonFactory.createParser(metricsInput);
```

Comment:
No, I didn't think that was worth messing with right now... seems like something we can add later if we ever actually run into an issue with it.  The parser works fine with multiple JSON objects in the file, so the only scenario where it would be an issue is if the last line in the file was corrupted, and that doesn't seem likely enough to be worth messing with in this pass.